python/ray/tune/suggest/hyperopt.py

```diff
@@ -361,15 +361,26 @@ def resolve_value(par: str, domain: Domain) -> Any:
                     return hpo.hp.randint(par, domain.upper)
             elif isinstance(domain, Categorical):
                 if isinstance(sampler, Uniform):
-                    return hpo.hp.choice(par, domain.categories)
+                    return hpo.hp.choice(par, [
+                        HyperOptSearch.convert_search_space(
+                            category, prefix=par)
+                        if isinstance(category, dict) else
+                        HyperOptSearch.convert_search_space(
+                            dict(enumerate(category)), prefix=f"par/{i}")
```

Comment:
What exactly is prefix used for/why is the variable par used as the prefix for dicts but as a string for lists? How is this reflected in the config dict?

Reply:
There's an error here, thanks - I'll fix it. Basically we need the prefix to avoid duplicate labels in hyperopt

Reply:
So `par` is always a string. Dicts then have their subkeys, so it will be `a/x`, a/y` anyway. But lists have to be converted to dicts in order to be convertable, and we then end up with `a/0`, `a/1`, etc. Resolving the values should be fine, as we end up with dicts with integer indices - they are not lists, but can still be accessed with numerical indices. See also the test for this.

Reply:
Got it, makes sense